datadog/client.go

```diff
@@ -102,24 +104,61 @@ func (c *Client) close() {
 	<-c.join
 }
 
+func socket(address string, sizehint int) (conn net.Conn, bufsize int, err error) {
```

Comment:
It would be great to have a comment here explaining why we need this method so that anyone coming along later understands why we're doing the extra work here.